x-pack/platform/plugins/shared/onechat/server/services/tools/esql/client.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { esqlToolProviderId } from '@kbn/onechat-common';
+import { logger } from 'elastic-apm-node';
+import { EsqlToolCreateRequest, EsqlToolCreateResponse } from '../../../../common/tools';
+import { esqlToolIndexName } from './storage';
+import { EsqlToolStorage } from './storage';
+
+export interface EsqlToolClient {
+  get(toolId: string): Promise<EsqlToolCreateResponse>;
+  list(): Promise<EsqlToolCreateResponse[]>;
+  create(esqlTool: EsqlToolCreateResponse): Promise<EsqlToolCreateResponse>;
+  update(toolId: string, updates: Partial<EsqlToolCreateRequest>): Promise<EsqlToolCreateResponse>;
+  delete(toolId: string): Promise<boolean>;
+}
+
+export const createClient = ({ storage }: { storage: EsqlToolStorage }): EsqlToolClient => {
+  return new EsqlToolClientImpl({ storage });
+};
+
+class EsqlToolClientImpl {
+  public readonly id = esqlToolProviderId;
+  private readonly storage: EsqlToolStorage;
+
+  constructor({ storage }: { storage: EsqlToolStorage }) {
+    this.storage = storage;
+  }
+
+  async get(id: string): Promise<EsqlToolCreateResponse> {
```

Comment:
NIT: having a `EsqlToolCreateResponse` as a return type for `get` and `list` APIs feels wrong. I would prefer something more explicit and clear such as `EsqlTool` or `EsqlToolDef`.